test/replication-luatest/gh_7486_deprecate_replication_synchro_timeout_test.lua

```diff
@@ -0,0 +1,164 @@
+    end)
+end
+
+g.test_option_not_available_if_new = function(cg)
+    cg.server:restart({
+        alias = 'new_behavior',
+        env = get_env('new'),
+    })
+    cg.server:exec(function()
+        -- Default value of replication_synchro_timeout with new behavior.
+        t.assert_equals(box.cfg.replication_synchro_timeout, 0)
+        local ok, err = pcall(box.cfg, { replication_synchro_timeout = 5, })
+        t.assert(not ok)
+        t.assert_equals(err.type, 'ClientError')
+        t.assert_equals(err.message, "Option 'replication_synchro_timeout' " ..
+            "is deprecated")
+        local ok, _ = pcall(box.cfg, { replication_synchro_timeout = 0, })
+        t.assert(ok)
+    end)
+end
+
+g.test_switch_option_to_new = function(cg)
+    cg.server:restart({
+        alias = 'old_behavior',
+        env = get_env('old'),
+    })
+    cg.server:exec(function()
+        t.assert_equals(box.cfg.replication_synchro_timeout, 5)
+        local switch_to_new = function()
+            require('compat').box_cfg_replication_synchro_timeout = 'new'
+        end
+        local exp_err =
+            "The compat option box_cfg_replication_synchro_timeout switches" ..
+            " to new only if box.cfg.replication_synchro_timeout is 0," ..
+            " but it is 5 now"
+        t.assert_error_msg_content_equals(exp_err, switch_to_new)
+        box.cfg{ replication_synchro_timeout = 0, }
+        local ok, _ = pcall(switch_to_new)
+        t.assert(ok)
+    end)
+end
+
+for _, case in ipairs({
+    {
+        'test_config_module_option_new_default',
+        compat_value = "'new'",
+        default = 0,
+    },
+    {
+        'test_config_module_option_old_default',
+        compat_value = "'old'",
+        default = 5,
+    },
+}) do
+    g[case[1]] = function(cg)
+        local dir = treegen.prepare_directory(cg, {}, {})
+        local file_config = [[
+            compat:
+              box_cfg_replication_synchro_timeout: %s
+
+            groups:
+              group-001:
+                replicasets:
+                  replicaset-001:
+                    instances:
+                      instance-001: {}
+        ]]
+        local config = string.format(file_config, case.compat_value)
+        treegen.write_script(dir, 'config.yaml', config)
+
+        local script = [[
+            print(box.cfg.replication_synchro_timeout)
+            os.exit(0)
+        ]]
+        treegen.write_script(dir, 'main.lua', script)
+
+        local env = {TT_LOG_LEVEL = 0}
+        local opts = {nojson = true, stderr = false}
+        local args = {'--name', 'instance-001', '--config', 'config.yaml',
+                      'main.lua'}
+        local res = justrun.tarantool(dir, env, args, opts)
+        t.assert_equals(res.exit_code, 0)
+        t.assert_equals(res.stdout, table.concat({case.default}, "\n"))
+    end
+end
+
+-- We cannot check that the timeout is truly infinite,
+-- but we can check that it is not 0.
+for _, case in ipairs({
```

Comment:
case[1] is the test name. Each of tests
```lua
{
    {
        'test_zero_value_does_not_rollback_immediately_if_new',
        compat_value = 'new',
    },
    {
        'test_zero_value_does_not_rollback_immediately_if_old',
        compat_value = 'old',
    },
}
```
was executed.
> it won't work on the new one

Why? In both old and new modes, I can set an infinite timeout, and check that it does not roll back the transaction immediately. Since before this there was a bug in which only in one mode the transaction was rolled back immediately, I decided to add a test for each of the modes.

Reply:
My bad, I missed, that you use `compat_value` in the test. No objections then